File: backend-python/app/modules/system/dto/menu_request.py
from pydantic import BaseModel, Field


class MenuSaveRequest(BaseModel):
    parent_id: int | None = Field(default=None, alias="parentId")
    name: str = Field(min_length=1, max_length=64)
    type: str = Field(min_length=1, max_length=16)
    path: str | None = Field(default=None, max_length=255)
    component: str | None = Field(default=None, max_length=255)
    permission_code: str | None = Field(default=None, alias="permissionCode", max_length=128)
    icon: str | None = Field(default=None, max_length=64)
    sort: int | None = None
    visible: int | None = Field(default=None, ge=0, le=1)
    status: int | None = Field(default=None, ge=0, le=1)

File: backend-rust/src/modules/system/dto/menu_request.rs
use crate::common::I64String;
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize, utoipa::ToSchema)]
#[serde(rename_all = "camelCase")]
pub struct MenuSaveRequest {
    pub parent_id: Option<I64String>,
    pub name: String,
    pub r#type: String,
    pub path: Option<String>,
    pub component: Option<String>,
    pub permission_code: Option<String>,
    pub icon: Option<String>,
    pub sort: Option<i32>,
    pub visible: Option<i32>,
    pub status: Option<i32>,
}
